### PR TITLE
Remove soft deprecated function `bookmarkableShinyApp`

### PR DESCRIPTION
**Pull request by chlebowa**
Closes [this](https://github.com/insightsengineering/teal/issues/756).

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -14,8 +14,10 @@
 * Updated examples to use `scda.2022`.
 * Added R session information into a link in the footer of `teal` applications.
 * Added data hashing step using `rlang` instead of `digest` package to calculate the hash (which has been moved from `teal.data` and `teal.slice`). There is now an explicit hashing check in the reproducible code output.
+* Removed deprecated function `root_modules`.
```
Comment (mhallal1) [inline NEWS.md:17]: The NEWS entry has been added here but was not removed from line 41, please remove it.

Comment (mhallal1) [reply NEWS.md:17]: and deprecated functions can be grouped in a single entry.

**Review by donyunardi — CHANGES_REQUESTED** [review]
I got this error message when trying to run a simple teal app:

```r
example("init", package="teal")
```

image

Which was caused by this line:
https://github.com/insightsengineering/teal/blob/07d17d133c253dbcad46dc01198eb613aec11327/R/module_teal.R#L210

Maybe we can just remove this `if` statement?

**Comment by chlebowa** [comment]
> (...) caused by this line:
> 
> https://github.com/insightsengineering/teal/blob/07d17d133c253dbcad46dc01198eb613aec11327/R/module_teal.R#L210
> 
> Maybe we can just remove this `if` statement?

I believe was part of the deprecated bookmarking mechanism. Removed altogether. Tests pass, example app works.

**Review by donyunardi — APPROVED** [review]
I saw this warning when adding teal.reporter card in exploratory app:
```r
[WARNING] Deprecated: --self-contained. use --embed-resources --standalone
```

I think this is pandoc related.

Otherwise, LGTM!

**Pull request merged**